libs/fills/src/lib/fills-manager.tsx

```diff
@@ -1,80 +1,92 @@
-import { fillsDataProvider } from './fills-data-provider';
+import { useCallback, useRef, useMemo } from 'react';
 import { useDataProvider } from '@vegaprotocol/react-helpers';
 import { AsyncRenderer } from '@vegaprotocol/ui-toolkit';
-import type { FillsVariables } from './__generated__/Fills';
-import type { FillFields } from './__generated__/FillFields';
+import { FillsTable } from './fills-table';
+import type { IGetRowsParams } from 'ag-grid-community';
+
+import { fillsDataProvider as dataProvider } from './fills-data-provider';
+import type { Fills_party_tradesPaged_edges } from './__generated__/Fills';
 import type { FillsSub_trades } from './__generated__/FillsSub';
-import isEqual from 'lodash/isEqual';
 
 interface FillsManagerProps {
   partyId: string;
 }
 
 export const FillsManager = ({ partyId }: FillsManagerProps) => {
   const gridRef = useRef<AgGridReact | null>(null);
-  const variables = useMemo<FillsVariables>(
-    () => ({
-      partyId,
-      pagination: {
-        last: 300,
-      },
-    }),
-    [partyId]
-  );
-  const update = useCallback((delta: FillsSub_trades[]) => {
-    if (!gridRef.current) {
-      return false;
-    }
-    const updateRows: FillFields[] = [];
-    const add: FillFields[] = [];
+  const dataRef = useRef<Fills_party_tradesPaged_edges[] | null>(null);
+  const totalCountRef = useRef<number | undefined>(undefined);
 
-    delta.forEach((d) => {
+  const update = useCallback(
+    ({ data }: { data: Fills_party_tradesPaged_edges[] }) => {
       if (!gridRef.current?.api) {
-        return;
-      }
-
-      const rowNode = gridRef.current.api.getRowNode(d.id);
-
-      if (rowNode) {
-        if (!isEqual(d, rowNode.data)) {
-          updateRows.push(d);
-        }
-      } else {
-        add.push(d);
+        return false;
       }
-    });
+      dataRef.current = data;
+      gridRef.current.api.refreshInfiniteCache();
+      return true;
+    },
+    []
+  );
 
-    if (updateRows.length || add.length) {
-      gridRef.current.api.applyTransactionAsync({
-        update: updateRows,
-        add,
-        addIndex: 0,
-      });
-    }
+  const insert = useCallback(
+    ({
+      data,
+      totalCount,
+    }: {
+      data: Fills_party_tradesPaged_edges[];
+      totalCount?: number;
+    }) => {
+      dataRef.current = data;
+      totalCountRef.current = totalCount;
+      return true;
+    },
+    []
+  );
 
-    return true;
-  }, []);
+  const variables = useMemo(() => ({ partyId }), [partyId]);
 
-  const { data, loading, error } = useDataProvider(
-    fillsDataProvider,
-    update,
-    variables
-  );
+  const { data, error, loading, load, totalCount } = useDataProvider<
+    Fills_party_tradesPaged_edges[],
+    FillsSub_trades[]
+  >({ dataProvider, update, insert, variables });
+  totalCountRef.current = totalCount;
+  dataRef.current = data;
 
-  const fills = useMemo(() => {
-    if (!data?.length) {
-      return [];
+  const getRows = async ({
```

Comment:
comment (non-blocking): ngl, i find this piece of code extremely hard to understand, and very not reacty. creating a function here to then be able to modify the result of another hook from inside of a child component without passing the data itself in feels just wrong to me, but again, i am not sure i understand the whole context of what this is trying to achieve. will need some time to go over it and unwrap things, i don't have any suggestions just yet on how to potentially improve this, but the general impression is that code just doesn't need to be this hard to read.